hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAccountStore.java

```diff
@@ -0,0 +1,169 @@
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for working with
+ * accounts.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail. This
+ * class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableAccountStore extends ReadableAccountStoreImpl {
+    /** The underlying data storage class that holds the account data. */
+    private final WritableKVState<EntityNumVirtualKey, Account> accountState;
+    /** The underlying data storage class that holds the aliases data built from the state. */
+    private final WritableKVState<String, EntityNumValue> aliases;
+
+    /**
+     * Create a new {@link WritableAccountStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAccountStore(@NonNull final WritableStates states) {
+        super(states);
+        requireNonNull(states);
+
+        this.accountState = states.get(TokenServiceImpl.ACCOUNTS_KEY);
+        this.aliases = states.get(TokenServiceImpl.ALIASES_KEY);
+    }
+
+    /**
+     * Persists a new {@link Account} into the state, as well as exporting its ID to the transaction
+     * receipt.
+     *
+     * @param account - the account to be added to modifications in state.
+     */
+    public void put(@NonNull final Account account) {
+        Objects.requireNonNull(account);
+        accountState.put(EntityNumVirtualKey.fromLong(account.accountNumber()), Objects.requireNonNull(account));
+    }
+
+    /**
+     * Persists a new alias linked to the account persisted to state
+     *
+     * @param alias - the alias to be added to modifications in state.
+     * @param accountNum - the account number to be added to modifications in state.
+     */
+    public void putAlias(@NonNull final String alias, @NonNull final long accountNum) {
+        Objects.requireNonNull(alias);
+        aliases.put(alias, new EntityNumValue(accountNum));
+    }
+
+    /** Commits the changes to the underlying data storage. */
+    public void commit() {
+        ((WritableKVStateBase) accountState).commit();
+        ((WritableKVStateBase) aliases).commit();
+    }
+
+    /**
+     * Returns the {@link Account} with the given number. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param accountID - the id of the Account to be retrieved.
+     */
+    @NonNull
+    public Optional<Account> get(final AccountID accountID) {
+        requireNonNull(accountID);
+        final var account = getAccountLeaf(accountID);
+        return Optional.ofNullable(account);
+    }
+
+    /**
+     * Returns the {@link Account} with the given number using {@link
+     * WritableKVState#getForModify(Object K)}}. If no such account exists, returns {@code
+     * Optional.empty()}
+     *
+     * @param id - the number of the account to be retrieved.
+     */
```

Comment:
A couple thoughts here:

Explaining the implementation of this method by including `WritableKVState#getForModify` leaks the implementation into the API contract of this class. Unless there's an explicit reason to include it, I would think it shouldn't be in here. 

```suggestion
    /**
     * Returns the {@link Account} with the given number. If no such account exists, returns {@code
     * Optional.empty()}
```

This doc is also a little confusing given the difference between the account ID and the account number. 
```suggestion
    /**
     * Returns the {@link Account} with the given ID. If no such account exists, returns {@code
     * Optional.empty()}
     *
     * @param id - the account ID of the account to be retrieved.
     */
```

Reply:
Done. Thanks